sdk/src/test/scala/com/horizen/validation/ConsensusValidatorOmmersTest.scala

```diff
@@ -144,106 +138,369 @@ class ConsensusValidatorOmmersTest
+    var slotsInEpoch: Int = 6
+    var history = mockHistory(slotsInEpoch)
+
+    val previousEpochNumber: ConsensusEpochNumber = ConsensusEpochNumber @@ 2
+    val currentEpochNumber: ConsensusEpochNumber = ConsensusEpochNumber @@ 3
 
-    // Test 1: Valid Ommers in correct order from the previous epoch as VerifiedBlock
     // Mock ommers
     val ommers: Seq[Ommer] = Seq(
-      getMockedOmmer(history.getTimeStampForEpochAndSlot(ConsensusEpochNumber @@ 2, ConsensusSlotNumber @@ 20)),
-      getMockedOmmer(history.getTimeStampForEpochAndSlot(ConsensusEpochNumber @@ 3, ConsensusSlotNumber @@ 9))
+      getMockedOmmer(history.getTimeStampForEpochAndSlot(previousEpochNumber, ConsensusSlotNumber @@ 5)), // quite slot - no impact on nonce calculation
+      getMockedOmmer(history.getTimeStampForEpochAndSlot(currentEpochNumber, ConsensusSlotNumber @@ 1)),
+      getMockedOmmer(history.getTimeStampForEpochAndSlot(currentEpochNumber, ConsensusSlotNumber @@ 5))
     )
 
     // Mock block with ommers
-    val verifiedBlockTimestamp = history.getTimeStampForEpochAndSlot(ConsensusEpochNumber @@ 3, ConsensusSlotNumber @@ 10)
-    var verifiedBlock: SidechainBlock = mock[SidechainBlock]
-    Mockito.when(verifiedBlock.id).thenReturn(verifiedBlockId)
+    val parentId: ModifierId = getRandomBlockId()
+    val parentInfo: SidechainBlockInfo = mock[SidechainBlockInfo]
+    val verifiedBlockTimestamp = history.getTimeStampForEpochAndSlot(currentEpochNumber, ConsensusSlotNumber @@ 6)
+    val verifiedBlock: SidechainBlock = mock[SidechainBlock]
     val header = mock[SidechainBlockHeader]
     Mockito.when(header.timestamp).thenReturn(verifiedBlockTimestamp)
     Mockito.when(verifiedBlock.header).thenReturn(header)
     Mockito.when(verifiedBlock.ommers).thenReturn(ommers)
 
+
+    Mockito.when(history.calculateNonceForNonGenesisEpoch(
+      ArgumentMatchers.any[ModifierId],
+      ArgumentMatchers.any[SidechainBlockInfo],
+      ArgumentMatchers.any[Seq[(VrfOutput, ConsensusSlotNumber)]])).thenAnswer(answer => {
+      val lastBlockIdInEpoch: ModifierId = answer.getArgument(0)
+      val lastBlockInfoInEpoch: SidechainBlockInfo = answer.getArgument(1)
+      val acc: Seq[(VrfOutput, ConsensusSlotNumber)] = answer.getArgument(2)
+
+      assertEquals("On calculate nonce: lastBlockIdInEpoch is different", parentId, lastBlockIdInEpoch)
+      assertEquals("On calculate nonce: lastBlockInfoInEpoch is different", parentInfo, lastBlockInfoInEpoch)
+      val expectedAcc: Seq[(VrfOutput, ConsensusSlotNumber)] = Seq(
+        (null, ConsensusSlotNumber @@ 5)
+      )
+      assertEquals("On calculate nonce: acc is different", expectedAcc, acc)
+
+      // Return nonce same as current epoch nonce
+      currentFullConsensusEpochInfo.nonceConsensusEpochInfo
+    })
+
+    var switchedEpochConsensusValidator = new ConsensusValidator {
+      /*override def verifyVrfProofAndHash(history: SidechainHistory, header: SidechainBlockHeader, message: VrfMessage): Unit = {
```

Comment:
do we need that comment?

Reply:
I just left it for further discussion. I guess, that this should be replaced somehow with VRFOuput check, but this is more difficult than for message.